src/Microsoft.AspNetCore.Mvc.Razor.Extensions/PageAttributeInjectionPass.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Extensions
+{
+    public class PageAttributeInjectionPass : RazorIRPassBase, IRazorIROptimizationPass
```

Comment:
Yeah. I guess I should call it `AssemblyAttributeInjectionPass` or something along those lines.